src/main/ruby/truffleruby/core/io.rb

```diff
@@ -1225,9 +1216,7 @@ def read_to_separator
         end
       end
 
-      str << @buffer.shift
-      str.chomp!(@separator) if @chomp
-      yield_string(str) { |y| yield y }
```

Comment:
It seems nice to replace other usages of `yield_string` too, especially since those actually cost an extra block+yield compared to `yield prepare_read_string(str) unless str.empty?`.

Reply:
I can do that. I was just trying to keep the scope limited.

Reply:
Let's do it since this usage was changed too, that way it will also avoid some duplication